services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crt/S3CrtResponseHandlerAdapter.java

```diff
@@ -63,14 +65,13 @@ public void onResponseHeaders(int statusCode, HttpHeader[] headers) {
         for (HttpHeader h : headers) {
             respBuilder.appendHeader(h.getName(), h.getValue());
         }
-
         respBuilder.statusCode(statusCode);
-        responseHandler.onHeaders(respBuilder.build());
-        responseHandler.onStream(responsePublisher);
     }
 
     @Override
     public int onResponseBody(ByteBuffer bodyBytesIn, long objectRangeStart, long objectRangeEnd) {
+        initiateResponseHandling(respBuilder.build());
```

Comment:
If there's no response payload, onResponseBody will not be invoked, so should we call initiateResponseHandling in `onFinished` for successful response as well?

Reply:
Ah right, makes sense.